src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -776,19 +776,24 @@ static int do_map(int argc, const char *argv[])
 
 static int do_unmap()
 {
+  
+  int r = 0;
+
   int nbd = open_device(devpath.c_str());
   if (nbd < 0) {
     cerr << "rbd-nbd: failed to open device: " << devpath << std::endl;
     return nbd;
   }
 
-  if (ioctl(nbd, NBD_DISCONNECT) < 0) {
-    cerr << "rbd-nbd: the device is not used" << std::endl;
+  r = ioctl(nbd, NBD_DISCONNECT);
+  if (r < 0) {
+      r = -errno;
+      cerr << "rbd-nbd: disconnect failed: " << cpp_strerror(-r) << std::endl;
```

Comment:
I am not sure I understand your example. If an nbd device is used by someone it should be in connected state. Then  NBD_DISCONNECT will succeed. Looking at the current code of the nbd driver [1] it looks like `nbd_disconnect` will return error (-EINVAL) only when the device is not connected. So for me:
```
rbd-nbd: the device is not used
```
looks nicer than
```
rbd-nbd: disconnect failed: (22) Invalid argument
```
Though, I don't have strong opinion. I just explain why it could be done this way initially (I am not the author) and why I don't consider this a bug. May be we will hear other people opinion before making a decision.

[1]  http://elixir.free-electrons.com/linux/latest/source/drivers/block/nbd.c#L710